src/complex.rs

```diff
@@ -920,4 +924,11 @@ mod test {
         assert!(::hash(&b) != ::hash(&c));
         assert!(::hash(&c) != ::hash(&a));
     }
+
+    #[test]
+    fn test_is_nan() {
+        assert!(!(_1_1i.is_nan()));
+        let a = Complex::new(f64::NAN, f64::NAN);
+        assert!(a.is_nan());
+    }
```

Comment:
Add test cases for `(NaN, _)` and `(_, NaN)` and IMHO will be ready to go.

Oh, and split it into several tests. `_1_1i.is_nan()` should be IMHO in different test case.